### DISCUSSION TITLE
Inquiring About Built-in Cluster Detection

### DISCUSSION THREAD

**Discussion by CUHKfreshman** [cosmograph-org/cosmos#52] [Q&A] [answered]
First of all, I would like to express my gratitude for your incredible work! Your algorithm is remarkably efficient.

While reviewing the code, I couldn't locate any specific details related to my query. I'm curious to know if there's an existing built-in method, or a relevant one, for explicitly returning the visualized clusters, or if there are any plans to introduce such a feature in the near future? Thank you very much :)

**Comment by rokotyan** [comment, marked as answer]
Thanks @CUHKfreshman!

Cosmos does not have a built-in method of network community detection, and we have no plans to add it at the moment.
I can suggest you to use the [networkanalysis-ts](https://github.com/neesjanvaneck/networkanalysis-ts) package that implements the [Leiden algorithm](https://doi.org/10.1038/s41598-019-41695-z), initially developed by [Vincent Traag](https://orcid.org/0000-0003-3170-3879), [Nees Jan van Eck](https://orcid.org/0000-0001-8448-4521), and [Ludo Waltman](https://orcid.org/0000-0001-8249-1752) (and we helped with porting it to TypeScript).

**Reply by CUHKfreshman** [reply]
Thanks for your helpful response! Your solution sounds great and I'm confident it will work for me. I appreciate you sharing your knowledge with me.

Thanks again!
